utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -40,8 +40,13 @@ pub(crate) enum ProcessedSysCallParams {
         value_type: ValueType,
         allowed_values: Option<SysCallParamAllowedValues>,
     },
-    MemoryArray,
-    MemoryPtrValue,
+    MemoryArray {
+        pointer_allowed_values: Option<SysCallParamAllowedValues>,
+        length_allowed_values: Option<SysCallParamAllowedValues>,
```

Comment:
That creates an issue with memory arrays. Rules for `ProcessedSysCallParams` must be enabled by desire in config for each sys-call separately. What I mean is that when you define a rule for `ParamType::Ptr` and `ParamType::Size`, it affects not only `Value`, `MemoryPtrValue`, but `MemoryArray`. Default implementation of instructions generation for `MemoryArray` is good enough - it chooses from the existing memory range. Your changes override that with constant values that aren't guaranteed to be in an existing wams memory range. Maybe, it will work for default memory config.

Reply:
What if we change `SysCallsParamsConfig` to work not with `ParamType` but with higher-level entities like memory arrays?

Reply:
`MemoryArray` is not a user-space type.